File: src/main/java/net/hectus/neobb/turn/default_game/flower/TAzureBluet.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import org.bukkit.block.Block;

import java.util.List;

public class TAzureBluet extends FlowerTurn {
    public TAzureBluet(NeoPlayer player) { super(player); }
    public TAzureBluet(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 7;
    }

    @Override
    public boolean unusable() {
        if (isDummy()) return true;
        return player.game.history().stream().noneMatch(t -> t instanceof TAzureBluet);
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Luck(Buff.BuffTarget.YOU, 20));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/flower/TAzureBluet.kt
package net.hectus.neobb.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TAzureBluet(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 7

    override fun unusable(): Boolean {
        if (isDummy()) return true
        return player!!.game.history.none { it is TAzureBluet }
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(Luck(20))
    }
}
